doc/kios.md

```diff
@@ -1 +1,871 @@
+404 | Not Found | User not found in database  
+500 | Internal Server Error | some un-handle error in server 
+200 | OK | OK
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data": {
+        "id": 1,
+        "kios_name": "mandala",
+        "address": jl. "bekasi timur raya no. 159 A",
+        "branches": [
+            {
+                "branch_code": "02629"
+            }
+        ]
+    }
+}
+```
+
+### Logic
+
+#### Validation
+- kios_name     : required and not empty
+- address       : required and not empty
+- branch_code   : required and not empty
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Request Payload : empty
+
+response HTTP Status Code : 400
+
+Response Payload : 
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "kios_name is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "kios_name is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Request payload :
+```
+{
+    "kios_name": ""
+}
+```
+
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "cdc-404",
+    "status_message": "kios_name is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
+
+Request Payload :
+```
+{
+    "kios_name": "mandala"
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "address is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 5
+
+Request Payload :
+```
+{
+    "kios_name": "mandala",
+    "address": ""
+}
+```
+ 
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "cdc-404",
+    "status_message": "address is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 6
+
+Request Payload :
+```
+{
+    "kios_name": "mandala",
+    "address": "jl. bekasi timur raya no. 159 A",
+}
+```
+ 
+Response HTTP Status Code : 400
+
+Response Payload:
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "branch_code is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 7
+
+Request Payload :
+```
+{
+    "kios_name": "mandala",
+    "address": "jl. bekasi timur raya no. 159 A",
+    "branch_code": ""
+}
+```
+ 
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "cdc-404",
+    "status_message": "branch_code is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 8
+
+Request Payload :
+```
+{
+    "kios_name": "mandala",
+    "address": "jl. bekasi timur raya no. 159 A",
+    "branch_code": "asal"
+}
+```
+ 
+Response HTTP Status Code : 400
+
+Response Payload:
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "branch_code is not found",
+    "data": null
+}
+```
+
```

Comment:
Add negative case for 403 forbidden when the brach_code is not owned by user login.